_docs/step-development-guideline.md

```diff
@@ -1,5 +1,45 @@
 # Step Development Guideline
 
+## Step dependencies
+
+In the context of Steps, dependency means an executable available in the PATH environment variable, which is executed by the step.
```

Comment:
It is technically not true. Dependency may be (among others) a library which is not executed nor included in `PATH`. For example `flutter` step depends on `libglu1-mesa` https://github.com/bitrise-io/bitrise-steplib/blob/master/steps/flutter/0.0.7/step.yml#L44

Reply:
@koral-- thanks for the review!
With this doc we define what a dependency means in a case of a step, not in general.
After some discussions, we agreed: step dependencies should include libraries, tools invoked by the step and also tools needed to perform the step.

The issue, which initially led to some disagreement is that the CLI uses `which` command to check if a given dependency is installed or not (`which` looks for the tool in the `PATH` env var). This way in case of a dependency, which is not available in the `PATH`, the CLI will call the appropriate install command regardless if that dependency is installed or not.